src/lib.rs

```diff
@@ -1099,6 +1099,40 @@ impl SymbolKind {
     pub const TYPE_PARAMETER: SymbolKind = SymbolKind(26);
 }
 
+impl Debug for SymbolKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
```

Comment:
Nice find, thanks. IMO these stringified representations (with the official LSP names) should definitely be exposed, because they are good enough for many clients.

I wondered if we should implement `Display` instead of `Debug` because at least kak-lsp shows this to the user.  Then again `_ => write!(f, "SymbolKind({})", self.0),` feel more at home in `Debug`, so I think it's fine.

In future it would be nice to have the same for the other 16 enum-turned-struct types.  Maybe someone can devise a proc-macro to generate the methods - we just need to convert `SOME_NAME` to `SomeName` everywhere.

Reply:
https://github.com/gluon-lang/lsp-types/pull/222